### ISSUE TITLE
CLI should check parameter name, and exit in error when encountering an unknown parameter

### ISSUE THREAD

**Issue by oupala** [FreshRSS/FreshRSS#2046]
When I enter an api password in the web interface, I get an error because of an error format. But there is no tip about a specific format.

When I set the api password with the cli, I get no error, only a success. With the same password. But in the end, the password does not work with EasyRSS.

What is the minimal size of the api password? If the password is shorter and is set via the cli, should it work or should it fail?

**Comment by oupala** [comment]
In the Web UI, it is not very clear to see that the api password is *also* constrained to 7 chars (as the main password.

**Comment by oupala** [comment]
As far as I have tested, creating a user via CLI does not set the api password.

Here is my command line:

```bash
{{ _freshrss.install_path }}/cli/create-user.php --user {{ _freshrss.end_user.login }} --password '{{ _freshrss.end_user.password }}' --api-password '{{ _freshrss.end_user.password }}' --language {{ _freshrss.locale }} --email {{ _freshrss.end_user.email }} --token {{ _freshrss.token }}
```

As you can see, I'm trying to set the api password: `--api-password '{{ _freshrss.end_user.password }}'` (all vars are infered by ansible).

But in the end, the config file shows that no password is set:

```config
  'passwordHash' => 'filled-with-a-hash',
  'apiPasswordHash' => '',
```

I can however set the api password by the following command line:

```bash
{{ _freshrss.install_path }}/cli/update-user.php --user myusername --api_password 'myapipassword'
```

So, as far as I understand, `cli/update-user.php` works while `cli/create-user.php` does not.

**Comment by Alkarex** [comment]
Please note that it is `--api_password` with an underscore, in both cases

**Comment by oupala** [comment]
It works way better with the right parameter name. Thanks!

The strange thing is that hashes are different event if passwords are the same:

```bash
--password '{{ _freshrss.end_user.password }}' --api-password '{{ _freshrss.end_user.password }}'
```

By the way, this could be an opportunity to rename this issue and to improve the CLI robustness by checking the parameter's name and by exiting in error if one parameter is not recognized. This would be clearly better that silently ignoring a parameter that is wrongly spelled... Don't you think?

**Comment by Alkarex** [comment]
Yes, hashes are unique, because of a random salt.
Mja, we could put some warnings

**Comment by aledeg** [comment]
@oupala Please review the related PR!

**Comment by oupala** [comment]
I'm not sure to understand what pull request you are talking about. Could you please be more precise?

**Comment by aledeg** [comment]
#2278

**Comment by oupala** [comment]
Ok I see. But I'm not involved in this pull request. I think you confused me with @Alkarex.

**Comment by Frenzie** [comment]
@oupala Not involved, but it fixes the bug you found. :-)

**Comment by aledeg** [comment]
@oupala I was asking if the PR solves your issue or if there is still something missing.

**Comment by aledeg** [comment]
@oupala the fix is now in dev branch. Does this solve your issue?

**Comment by oupala** [comment]
Ok, I get it now. Sorry for being so long to understand...

I'll test it as soon as possible. Thanks for the notifications!

**Comment by aledeg** [comment]
No problem!

**Comment by oupala** [comment]
As far as I have tested merge request #2278, it fixes this bug.

Thanks @aledeg for fixing this.

I still have 2 remarks:

- it is very strange to get syntax inconsistency in parameter name: `--api_password` seems not natural as we usually use dash *or* underscore, but not both of them, we should convert all parameter to dash syntax: `--api-password`
- it is also very strange to see that installing FreshRSS on an already installed FreshRSS returns an error, following idempotency we should more return a `0` return code:

```bash
>php cli/do-install.php --default_user myuser --language fr --api-enabled --db-type sqlite
FreshRSS looks to be already installed! Please use `./cli/reconfigure.php` instead.
>echo $?
1
```

**Comment by aledeg** [comment]
Ping @Alkarex

**Comment by Alkarex** [comment]
The options name are with underscore, to match the values of https://github.com/FreshRSS/FreshRSS/blob/master/config.default.php and https://github.com/FreshRSS/FreshRSS/blob/master/config-user.default.php . It is not always super consistent.
`db` is special because it is to map an array of options (with the idea of possibly automating some mapping aspects, thanks to naming conventions).
I am not against changing, but then back-compatibility needs to be ensured somehow.

**Comment by Alkarex** [comment]
P.S. Let's open another issue if desired, as the main one here seems to be fixed by @aledeg , and can be included in the upcoming FreshRSS 1.14.0
